html_text/html_text.py

```diff
@@ -39,10 +41,33 @@ def parse_html(html):
     return lxml.html.fromstring(html.encode('utf8'), parser=parser)
 
 
-def selector_to_text(sel):
+_whitespace = re.compile(r'\s+')
+_has_trailing_whitespace = re.compile(r'\s$').search
+_has_punct_after = re.compile(r'^[,:;.!?"\)]').search
+_has_punct_before = re.compile(r'\($').search
+
+
+def selector_to_text(sel, guess_punct_space=True):
     """ Convert a cleaned selector to text.
+    See html_text.extract_text docstring for description of the approach and options.
     """
-    return sel.xpath('normalize-space()').extract_first('')
+    if guess_punct_space:
+
+        def fragments():
+            prev = None
+            for text in sel.xpath('//text()').extract():
```

Comment:
That's a great idea, thanks @redapple  - I'd like to also make it possible to pass selectors via the public interface.